formats/json/commonMain/src/kotlinx/serialization/json/Json.kt

```diff
@@ -313,8 +314,30 @@ public class JsonBuilder internal constructor(json: Json) {
      */
     public var useAlternativeNames: Boolean = json.configuration.useAlternativeNames
 
+    /**
+     * Specifies [JsonNamingStrategy] that should be used for all properties in classes for serialization and deserialization.
+     *
+     * `null` by default.
+     *
+     * This strategy is applied for all entities that have [StructureKind.CLASS].
+     */
+    public var namingStrategyForProperties: JsonNamingStrategy? = json.configuration.namingStrategyForProperties
+
+    /**
+     * Specifies [JsonNamingStrategy] that should be used for all enum instances for serialization and deserialization.
+     *
+     * `null` by default.
+     *
+     * This strategy is applied for all entities that have [SerialKind.ENUM].
+     */
+    public var namingStrategyForEnums: JsonNamingStrategy? = json.configuration.namingStrategyForEnums
```

Comment:
just interested: why to have 2 separate properties, as it's fair easy to handle this case inside `JsonNamingStrategy` like:
```kotlin
JsonNamingStrategy { descriptor, elementIndex, serialName -> 
  if(descriptor.kind == SerialKind.ENUM) serialName
  else JsonNamingStrategy.SnakeCase.serialNameForJson(descriptor, elementIndex, serialName)
}
```
And even introduce something like (name TBD):
```kotlin
fun JsonNamingStrategy.ifEnum(enumStrategy: JsonNamingStrategy): JsonNamingStrategy
```
if this use case is very often.

Reply:
I think it would be easier for users to configure them without going deep into descriptors and kinds since most of the strategies simply perform string operations.

But this is a draft design, maybe it's possible to come up with something that's better

Reply:
My main consideration with the separate options is that there are many more SerialKinds that one may want to support a naming strategy for. Doing it this way doesn't quite scale.

Reply:
After some consideration and discussions, we've decided that enums parsing are an entirely different feature from naming strategy: while strategy changes **names**, enum parsing should affect **values**. Besides, there are some cases when strategy for enums is not enough: e.g. you want to parse both `Foo` and `foo` into `FOO`, like Jackson's CASE_INSENSETIVE_ENUMS.

Therefore, enum support would be removed entirely from this PR. I'll reopen #209 for it.